src/NHibernate/Logging.cs

```diff
@@ -272,36 +358,158 @@ public bool IsEnabled(InternalLogLevel logLevel)
+		public void Debug(object message, Exception exception) => _this.Debug(exception, message?.ToString());
+
+		// Since v5.1
+		[Obsolete("Please use Debug(string, params object[]) INHibernateLogger extension method instead.")]
+		public void DebugFormat(string format, params object[] args) => _this.Debug(format, args);
+
+		// Since v5.1
+		[Obsolete("Please use Info(string, params object[]) INHibernateLogger extension method instead.")]
+		public void Info(object message) => _this.Info(message?.ToString());
+
+		// Since v5.1
+		[Obsolete("Please use Info(string, params object[]) INHibernateLogger extension method instead.")]
+		public void Info(string message) => _this.Info(message, default(object[]));
+
+		// Since v5.1
+		[Obsolete("Please use Info(Exception, string, params object[]) INHibernateLogger extension method instead.")]
+		public void Info(object message, Exception exception) => _this.Info(exception, message?.ToString());
+
+		// Since v5.1
+		[Obsolete("Please use Info(string, params object[]) INHibernateLogger extension method instead.")]
+		public void InfoFormat(string format, params object[] args) => _this.Info(format, args);
+
+		// Since v5.1
+		[Obsolete("Please use Warn(string, params object[]) INHibernateLogger extension method instead.")]
+		public void Warn(object message) => _this.Warn(message?.ToString());
+
+		// Since v5.1
+		[Obsolete("Please use Warn(string, params object[]) INHibernateLogger extension method instead.")]
+		public void Warn(string message) => _this.Warn(message, default(object[]));
+
+		// Since v5.1
+		[Obsolete("Please use Warn(Exception, string, params object[]) INHibernateLogger extension method instead.")]
+		public void Warn(object message, Exception exception) => _this.Warn(exception, message?.ToString());
+
+		// Since v5.1
+		[Obsolete("Please use Warn(string, params object[]) INHibernateLogger extension method instead.")]
+		public void WarnFormat(string format, params object[] args) => _this.Warn(format, args);
+	}
+
+	/// <summary>
+	/// Factory building loggers which log nothing.
+	/// </summary>
 	public class NoLoggingLoggerFactory: INHibernateLoggerFactory
 	{
-		private static readonly IInternalLogger2 Nologging = new NoLoggingInternalLogger();
-		public IInternalLogger2 LoggerFor(string keyName)
+		private static readonly INHibernateLogger Nologging = new NoLoggingInternalLogger();
+		/// <inheritdoc />
+		public INHibernateLogger LoggerFor(string keyName)
 		{
 			return Nologging;
 		}
 
-		public IInternalLogger2 LoggerFor(System.Type type)
+		/// <inheritdoc />
+		public INHibernateLogger LoggerFor(System.Type type)
 		{
 			return Nologging;
 		}
 	}
 
-	public class NoLoggingInternalLogger: IInternalLogger2
+#pragma warning disable 618 // NHibernateLoggerBase is obsolete, to be removed in a upcoming major version
+	/// <summary>
+	/// Logger which logs nothing.
+	/// </summary>
+	public class NoLoggingInternalLogger: NHibernateLoggerBase, INHibernateLogger
```

Comment:
Adding a base class or changing an interface for a more derived one is not a breaking change.